src/components/JUPTable/JUPTable.tsx

```diff
@@ -0,0 +1,184 @@
+import React from "react";
+import {
+  Box,
+  Paper,
+  styled,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TablePagination,
+  TableRow,
+  TableSortLabel,
+  Typography,
+} from "@mui/material";
+import SLink from "components/SLink";
+import { Data, IHeadCellProps } from "views/Dashboard/components/Widgets/TransactionsWidget/TransactionsWidget";
+import { ITransaction } from "types/NXTAPI";
+import { visuallyHidden } from "@mui/utils";
+import { TableRowsPerPageOptions } from "utils/common/constants";
+
+// might still want to use this concept
+// function createWidgetRow(date: string, qty: number, toFrom: string): Data {
+//   return {
+//     date,
+//     qty,
+//     toFrom,
+//   };
+// }
+
+interface ITableTitleProps {
+  title: string;
+}
+const TableTitle: React.FC<ITableTitleProps> = ({ title }) => {
+  return (
+    <>
+      {
+        <TitleText variant="h6" id="tableTitle">
+          <SLink href="/transactions">{title}</SLink>
+        </TitleText>
+      }
+    </>
+  );
+};
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = "asc" | "desc";
+
+function getComparator<Key extends keyof ITransaction>(
+  order: Order,
+  orderBy: Key
+): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  return order === "desc" ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
```

Comment:
instead of having to do this extra ternary here, it seems as tho the orderBy value is 'asc' or 'desc'.

this can be used inside descendingComparator function (suggestion: rename if you make my suggested change) to set a value which can be multiplied to the current -1 and 1 values being returned. -1 * -1 = 1 which would flip the meaning and thus should flip from descending to ascending.

Reply:
this sorting code is something I wanted to have a larger discussion with you about so i'm going to leave it alone for now because it's possible it needs to change beyond what you're suggesting here